tests/unit_tests/test_gameconnection.py

```diff
@@ -438,56 +443,143 @@ async def test_handle_action_GameEnded_ends_game(
     game.on_game_end.assert_called_once()
 
 
-async def test_handle_action_OperationComplete(ugame: Game, game_connection: GameConnection, database):
-    """
-        Sends an OperationComplete action to handle action and verifies that
-    the `coop_leaderboard` table is updated accordingly.
+@pytest.mark.parametrize(
+    "primary,secondary",
+    [(1, 1), (True, True), ("True", "True"), ("true", "true")],
+)
+async def test_handle_action_OperationComplete(
+    primary,
+    secondary,
+    coop_game: CoopGame,
+    game_connection: GameConnection,
+    database,
+):
+    coop_game.id = 1  # reuse existing corresponding game_stats row
+    coop_game.map_file_path = "maps/prothyon16.v0005.zip"
+    game_connection.game = coop_game
+    time_taken = "09:08:07.654321"
 
-    Requires that the map from `game.map_file_path` exists in the database.
-    """
+    await game_connection.handle_action(
+        "OperationComplete", [primary, secondary, time_taken]
+    )
 
-    ugame.map_file_path = "maps/prothyon16.v0005.zip"
-    ugame.validity = ValidityState.COOP_NOT_RANKED
-    game_connection.game = ugame
+    async with database.acquire() as conn:
+        result = await conn.execute(
+            select(
+                [coop_leaderboard.c.secondary, coop_leaderboard.c.gameuid]
+            ).where(coop_leaderboard.c.gameuid == coop_game.id),
+        )
 
-    secondary = 1
+        row = await result.fetchone()
+        assert (row[0], row[1]) == (1, coop_game.id)
+
+
+@pytest.mark.parametrize("primary", [0, False, "False", "false"])
```

Comment:
Don't wanna parametrize secondary here?

Reply:
The message handler bails before it does anything with `secondary` if `primary` is falsy, so there didn't seem much point?